### PR TITLE
ENHANCE: Make Config a singleton

### PR DESCRIPTION
**Pull request by zacbrannelly**
Example:
```python
from surround import has_config

@has_config
def some_function(config=None):
    config.read_from_dict({"override": False})

@has_config(name="local_config")
def some_other_function(local_config=None):
    assert(local_config["override"]) == False)
```

Also can be used like this if preferred:
```python
config = Config.instance()
```

Can also be used to override:
```python
@has_config(filename="override.yaml")
def some_function(config=None):
    ...
```

### CODE REVIEW

File: templates/new/web_main.py.txt [hunk 1]
```diff
@@ -17,8 +17,8 @@ assemblies = [
         .set_estimator(Baseline())
 ]
 
-def main():
-    config = Config(auto_load=True)
+@has_config
+def main(config):
```
Comment (mpej) [inline templates/new/web_main.py.txt]: Set default value of config to None, main is called with no arguments at bottom of file.

Comment (zacbrannelly) [reply templates/new/web_main.py.txt]: Done 👍

**Comment by ucokzeko** [comment]
@zacbrannelly Can you rebase this please? Thanks

**Comment by ucokzeko** [comment]
Just a recap from discussion with @zacbrannelly 
We will be trying to add an option to load other config files as an optional parameter of `@has_config`

**Pull request merged**